app/models/user.rb

```diff
@@ -35,30 +35,31 @@ def dislike(post)
   end
 
   def friends
-    friends_array = received_requests.map do |u|
-      u.sender if u.confirmed
-    end
-    friends_array += sent_requests.map do |u|
-      u.receiver if u.confirmed
-    end
-    friends_array.compact
+    friendships = sent_requests.includes(:receiver).where(confirmed: true).references(:users)
+    friendships.map(&:receiver)
   end
 
   def pending_friends
-    sent_requests.map { |u| u.receiver unless u.confirmed }.compact
+    requests = sent_requests.includes(:receiver).where(confirmed: false).references(:users)
+    requests.map(&:receiver)
   end
 
   def friend_requests
-    received_requests.map { |u| u.sender unless u.confirmed }.compact
+    requests = received_requests.includes(:sender).where(confirmed: false).references(:users)
+    requests.map(&:sender)
   end
 
   def confirm_friend(user)
```

Comment:
[OPTIONAL]

This could have easily been done in your friendship model.
Like the following:
```ruby
def confirm_friend
  self.update_attributes(confirmed: true)
  Friendship.create!(receiver_id: self.sender_id,
                  sender_id: self.receiver_id,
                  confirmed: true)
end
```
To create the duplicate rows. However, your way wax different and that is good too.

Creating it on friendship meant that you'd have only really needed to call this method on the friendship when the accept request was created and it seems also to be less verbose. Also, line 53 which is another query would be eliminated.